examples/platform/esp32/PigweedLogger.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "FreeRTOS.h"
+#include "semphr.h"
+#include <pw_hdlc/encoder.h>
+#include <pw_stream/sys_io_stream.h>
+
+namespace PigweedLogger {
+namespace {
+
+constexpr uint8_t kLogHdlcAddress = 1;   // Send log messages to HDLC address 1 (other than RPC communication)
+constexpr size_t kWriteBufferSize = 128; // Buffer for constructing HDLC frames
+
+SemaphoreHandle_t esp_log_mutex;
+
+pw::stream::SysIoWriter sWriter;
+size_t sWriteBufferPos;
+char sWriteBuffer[kWriteBufferSize];
+
+void send()
```

Comment:
All of `sWriter`, `sWriteBufferPos`, `sWriteBuffer`, and `send()` require mutex protection for correct operation.  Declaring these static would prevent inadvertent manipulation from other files.

Reply:
Done